broadinstitute/pearl
# AR-105 docker participant api

This configures the participant API and UI projects to match the work we did on the admin API to allow the UI to be served via the Spring app.  The one difference is that an environment variable "SWAGGER_ENABLED" is used for the participant-api to determine whether to serve a swagger ui page.  This variable defaults to false, but can be turned on if a developer wants to use the swagger to debug.

This also adds an auto-populator to api-admin.  It looks at the DB, and if there are no admin users, it runs the base setup script.  The admin users have now been expanded to the whole team.

TO TEST:
1. run `./local-dev/run_postgres.sh start`
2. (re)start your ApiAdminApp
3. Confirm the 7 admin users get created on startup
4. run `./gradlew api-participant:copyWebApp`
5. start the ApiParticipantApp
6. confirm going to localhost:8081 serves the participant UI static assets
